ProjectFiles/Assets/Scripts/Network/Server.cs

```diff
@@ -87,35 +94,78 @@ public void HandleNetworkEvents()
                         {
                             var readerContext = default(DataStreamReader.Context);
                             var ev = (ClientNetworkEvent) reader.ReadByte(ref readerContext);
-                            HandleEvent(connections[i], endpoint, ev, reader, readerContext);
+                            HandleEvent(connections[i], endpoint, ev, reader, readerContext, i);
                             break;
                         }
                         case NetworkEvent.Type.Disconnect:
                             Debug.Log($"Server: {endpoint.IpAddress()}:{endpoint.Port} disconnected.");
                             connections[i] = default(NetworkConnection);
+
+                            var playerID = connectionPlayerIDs[i];
```

Comment:
Same thing as above.

```suggestion
                            var playerID = connectionPlayerIDs[connections[i].InternalId];
```